### PR TITLE
Merge Multiselect list

### LINKED ISSUE #12: Multi-Select List
https://github.com/scality/pensieve-front/blob/master/src/react/ui-elements/MultipleSelect.jsx

State 1
Screen Shot 2019-04-18 at 2 28 04 PM

State 2
Screen Shot 2019-04-18 at 2 27 33 PM

State 3
Screen Shot 2019-04-18 at 2 27 21 PM

------------------------------
New Design:

image

image

### PR DESCRIPTION
**Pull request by carlito-scality**
**Component**:Multiselect

**Description**:
- Improve and merge this PR https://github.com/scality/core-ui/pull/96

**Design**:
image

image

image

**Breaking Changes**:

- [] Breaking Changes

---

Closes: https://github.com/scality/core-ui/issues/12

### CODE REVIEW

**Comment by carlito-scality** [comment]
@edumoreira1506 Hello :) I created a this PR to improve this component based on what you have done. I reused the existing core-ui components like Button, Select. I also optimized the design of this component by refactoring it to cover the most use cases as possible. Let me know if you have any questions ;)

**Comment by edumoreira1506** [comment]
This is awesome! Nice!!!

File: src/lib/components/multiselect/MultiSelect.component.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+};
+
+export type OptionProps = {
+  label: string,
+  value: string | number
+};
+
+export type SearchProps = {
+  placeholder?: string,
+  options: Array<OptionProps>, // "Content of option"
```
Comment (MonPote) [inline src/lib/components/multiselect/MultiSelect.component.js]: I guess "The options displayed in search suggestion" is a bit more explicit IMO

File: src/lib/components/multiselect/MultiSelect.component.js [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+type MultiSelectProps = {
+  title?: string,
+  items: Array<ItemProps>,
+  search?: SearchProps,
+  onItemRemove?: (any, any) => void //on item remove button click
+};
+
+const MultiSelectContainer = styled.div`
+  padding: ${defaultTheme.padding.base};
```
Comment (MonPote) [inline src/lib/components/multiselect/MultiSelect.component.js]: When the title is not present is it possible to put this padding at 0 ?

File: src/lib/components/multiselect/MultiSelect.component.js [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    onItemRemove,
+    onSelect,
+    isFavorite,
+    onFavoriteClick
+  } = props;
+
+  return (
+    <MultiSelectItemContainer className="sc-multi-select-item">
+      <MultiSelectItemLeft className="sc-multi-select-item-left">
+        {typeof selected !== "undefined" && onSelect && (
```
Comment (MonPote) [inline src/lib/components/multiselect/MultiSelect.component.js]: is this more accurate ?
`typeof selected !== "boolean"` ?

File: src/lib/components/multiselect/MultiSelect.component.js [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+          <CheckBox
+            checked={selected}
+            onChange={event => onSelect(label, event)}
+          />
+        )}
+        {typeof isFavorite !== "undefined" && onFavoriteClick && (
```
Comment (MonPote) [inline src/lib/components/multiselect/MultiSelect.component.js]: Same as `selected` but in this case should we rename it to `isSelected` for consistency ?

**Review by MonPote — APPROVED** [review]

**Comment by carlito-scality** [comment]
@bert-e approve

**Pull request merged**